src/common/proto.ts

```diff
@@ -0,0 +1,66 @@
+// Copied from firebase-tools/src/gcp/proto
+
+/**
+ * A type alias used to annotate interfaces as using a google.protobuf.Duration.
+ * This type is parsed/encoded as a string of seconds + the "s" prefix.
+ */
+export type Duration = string;
+
+/** Get a google.protobuf.Duration for a number of seconds. */
+export function durationFromSeconds(s: number): Duration {
+  return `${s}s`;
+}
+
+/**
+ * Utility function to help copy fields from type A to B.
+ * As a safety net, catches typos or fields that aren't named the same
+ * in A and B, but cannot verify that both Src and Dest have the same type for the same field.
+ */
+export function copyIfPresent<Src, Dest>(
+  dest: Dest,
+  src: Src,
+  ...fields: Array<keyof Src & keyof Dest>
+) {
+  for (const field of fields) {
+    if (!Object.prototype.hasOwnProperty.call(src, field)) {
+      continue;
+    }
+    dest[field] = src[field] as any;
+  }
+}
+
+export function renameIfPresent<Src, Dest>(
+  dest: Dest,
+  src: Src,
+  destField: keyof Dest,
+  srcField: keyof Src,
+  converter: (from: any) => any = (from: any) => {
+    return from;
+  }
+) {
+  if (!Object.prototype.hasOwnProperty.call(src, srcField)) {
+    return;
+  }
+  dest[destField] = converter(src[srcField]);
+}
+
+export function serviceAccountFromShorthand(
```

Comment:
This doesn't really have anything to do with protos, so I wonder if this should be elsewhere or the file should be named something else.